in_toto/runlib.py

```diff
@@ -310,49 +311,74 @@ def create_link_metadata(link_name, materials_dict={}, products_dict={},
+    Calls function to run command passed as link_cmd_args argument, storing
+    its materials, by-products and return value, and products into a link
+    metadata file. The link metadata file is signed with the passed key and
+    stored to disk.
 
   <Arguments>
-    link_name:
-            A unique name to relate link metadata with a step defined in the
-            layout.
-    materials_list:
-            A list of file or directory paths used as materials for
-            the link command.
-    products_list:
-            A list of file or directory paths used as materials for
-            the link command.
+    name:
+            A unique name to relate link metadata with a step or inspection
+            defined in the layout.
+    material_list:
+            List of file or directory paths that should be recorded as
+            materials.
+    product_list:
+            List of file or directory paths that should be recorded as
+            products.
     link_cmd_args:
             A list where the first element is a command and the remaining
             elements are arguments passed to that command.
-    record_byproducts:
+    key: (optional)
+            Private key to sign link metadata.
+            Format is ssl_crypto.formats.KEY_SCHEMA
+    record_byproducts: (optional)
             A bool that specifies whether to redirect standard output and
             and standard error to a temporary file which is returned to the
             caller (True) or not (False).
 
   <Exceptions>
-    TBA (see https://github.com/in-toto/in-toto/issues/6)
+    None.
 
   <Side Effects>
-    - Calls function to record materials.
-    - Calls function to execute link command.
-    - Calls function to record products.
-    - Calls function to create Link object.
+    Writes newly created link metadata file to disk "<name>.link"
 
   <Returns>
-    A Link metadata object
+    Newly created Link object
   """
-  materials_dict = record_artifacts_as_dict(materials_list)
+
+  fn = FILENAME_FORMAT.format(step_name=name)
+
+  log.doing("Running '{}'...".format(name))
+
+  if material_list:
+    log.doing("Recording materials '{}'...".format(", ".join(material_list)))
+  materials_dict = record_artifacts_as_dict(material_list)
+
+  log.doing("Running command '{}'...".format(" ".join(link_cmd_args)))
   byproducts, return_value = execute_link(link_cmd_args, record_byproducts)
-  products_dict = record_artifacts_as_dict(products_list)
-  return create_link_metadata(link_name, materials_dict, products_dict,
+
+  if product_list:
+    log.doing("Recording products '{}'...".format(", ".join(product_list)))
+  products_dict = record_artifacts_as_dict(product_list)
+
+  log.doing("Creating link metadata...")
+  link = create_link_metadata(name, materials_dict, products_dict,
       link_cmd_args, byproducts, return_value)
 
+  if key:
+    log.doing("Signing link metadata with key '{:.8}...'...".format(key["keyid"]))
+    link.sign(key)
```

Comment:
Correct. We should check on top of the function that the key is properly formatted.